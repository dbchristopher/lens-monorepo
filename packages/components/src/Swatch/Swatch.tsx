import * as React from 'react';
import { BoxProps } from '../Box/Box';
import Box from '../Box';

export interface SwatchProps extends BoxProps<HTMLDivElement> {
  /**
   * The background color to display on the swatch.
   */
  color?: string;
  /**
   * Swatch height.
   */
  height?: string;
  /**
   * Swatch width.
   */
  width?: string;
}

const InternalSwatch: React.FC<SwatchProps> = ({
  color,
  width,
  height,
  ...props
}) => {
  return (
    <Box
      border="1px solid"
      borderColor="palette.charcoal300"
      borderRadius="medium"
      bg={color || 'white'}
      width={width || '28px'}
      height={height || '28px'}
      {...props}
    />
  );
};

const Swatch = React.forwardRef<HTMLElement, SwatchProps>(
  (props: SwatchProps, ref) => {
    return <InternalSwatch {...props} innerRef={ref} />;
  }
);

export default Swatch;
