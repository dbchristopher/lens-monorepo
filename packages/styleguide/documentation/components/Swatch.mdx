```js noeditor
const StatusAndResources = require('../../../../styleguide_components/StatusAndResources')
  .StatusAndResources

;<StatusAndResources
  status="experimental"
  figmaURL="https://www.figma.com/file/h7RYPRCSlz3k8fLzEMRSzy/Color-Wheel?node-id=83%3A3"
  githubURL="https://github.com/looker/lens/blob/master/src/components/Colors/Swatch.tsx"
  feedbackTitle="Swatch Component Feedback"
/>
```

### Swatch

This simple component is passed a color and displays that color. The default `Swatch` is a 28px by 28px square, but can be passed `width` and `height` props.

```js
import { Swatch } from './Swatch'
;<Swatch color="purple" />
```
