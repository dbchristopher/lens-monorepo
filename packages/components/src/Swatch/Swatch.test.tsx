import 'jest-styled-components'
import * as React from 'react'
import { assertSnapshot } from '../../../../test/utils/snapshot'
import { Swatch } from './Swatch'

test('Default Swatch', () => {
  assertSnapshot(<Swatch />)
})

test('Swatch with hex value passed', () => {
  assertSnapshot(<Swatch color="#4c6670" />)
})

test('Swatch with width and height set', () => {
  assertSnapshot(<Swatch color="blue" width="50px" height="25px" />)
})
