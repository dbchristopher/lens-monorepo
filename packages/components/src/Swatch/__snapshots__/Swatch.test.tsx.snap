// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Default Swatch 1`] = `
.c0 {
  border: 0;
  box-sizing: border-box;
  font: inherit;
  font-size: 100%;
  margin: 0;
  padding: 0;
  vertical-align: baseline;
  -webkit-animation: none;
  animation: none;
  border: 1px solid;
  border-radius: 0.25rem;
  border-color: #C1C6CC;
  background-color: white;
  height: 28px;
  width: 28px;
}

<div
  className="c0"
/>
`;

exports[`Swatch with hex value passed 1`] = `
.c0 {
  border: 0;
  box-sizing: border-box;
  font: inherit;
  font-size: 100%;
  margin: 0;
  padding: 0;
  vertical-align: baseline;
  -webkit-animation: none;
  animation: none;
  border: 1px solid;
  border-radius: 0.25rem;
  border-color: #C1C6CC;
  background-color: #4c6670;
  height: 28px;
  width: 28px;
}

<div
  className="c0"
/>
`;

exports[`Swatch with width and height set 1`] = `
.c0 {
  border: 0;
  box-sizing: border-box;
  font: inherit;
  font-size: 100%;
  margin: 0;
  padding: 0;
  vertical-align: baseline;
  -webkit-animation: none;
  animation: none;
  border: 1px solid;
  border-radius: 0.25rem;
  border-color: #C1C6CC;
  background-color: blue;
  height: 25px;
  width: 50px;
}

<div
  className="c0"
/>
`;
